src/discordbot.ts

```diff
@@ -89,10 +89,7 @@ export class DiscordBot {
     let chan;
     let embed;
     const mxClient = this.bridge.getClientFactory().getClientAs();
-    return this.LookupRoom(guildId, channelId).then((channel) => {
-      chan = channel;
-      return mxClient.getProfileInfo(event.sender);
-    }).then((profile) => {
+    return this.LookupRoom(guildId, channelId).then((profile) => {
```

Comment:
The variables don't match here. `LookupRoom` resolves to a channel, not a profile.